File: src/lightnap-ng/src/app/core/backend-api/static-content-format.ts
import { ListItem } from "@core/models";
import { StaticContentFormat } from "./models";

export const StaticContentFormatListItems: Array<ListItem<StaticContentFormat>> = [
  new ListItem<StaticContentFormat>(StaticContentFormat.Html, "HTML", "Render as HTML."),
  new ListItem<StaticContentFormat>(StaticContentFormat.Markdown, "Markdown", "Render as Markdown."),
  new ListItem<StaticContentFormat>(StaticContentFormat.PlainText, "Plain Text", "Render as plain text."),
];

File: src/LightNap.Core/StaticContents/Enums/StaticContentFormat.cs
namespace LightNap.Core.StaticContents.Enums
{
    /// <summary>
    /// Supported CMS content formats.
    /// </summary>
    public enum StaticContentFormat
    {
        /// <summary>
        /// Renders as HTML.
        /// </summary>
        Html = 0,

        /// <summary>
        /// Renders as Markdown.
        /// </summary>
        Markdown = 1,

        /// <summary>
        /// Renders as plaintext (or used in other ways).
        /// </summary>
        PlainText = 2,
    }
}
